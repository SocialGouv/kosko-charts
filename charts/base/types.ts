//

export interface GlobalEnvironment {
  annotations?: Record<string, string>;
  domain: string;
  labels?: Record<string, string>;
  namespaceName: string;
  subdomain: string;
  subdomainSeparator: string;
}

export type AppComponentEnvironment = {
  containerPort: number;
  imageName: string;
  imageTag: string;
  labels?: Record<string, string>;
  limits?: { cpu: string; memory: string };
  name?: string;
  requests?: { cpu: string; memory: string };
  servicePort: number;
  subdomain?: string;
};

export type NamespaceComponentEnvironment = {
  annotations?: Record<string, string>;
  enabled: boolean;
  labels?: Record<string, string>;
};
