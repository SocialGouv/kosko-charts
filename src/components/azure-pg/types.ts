export interface PgParams {
  database: string;
  host: string;
  user: string;
  password: string;
  name: string;
  secretRefName: string;
}
