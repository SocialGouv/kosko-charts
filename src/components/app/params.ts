import { GlobalEnvironment } from "@socialgouv/kosko-charts/types";
import { Integer } from "@socialgouv/kosko-charts/utils/Integer";
import { NonEmptyString } from "@socialgouv/kosko-charts/utils/NonEmptyString";
import { pipe } from "fp-ts/lib/pipeable";
import * as D from "io-ts/lib/Decoder";

export const AppComponentParams = pipe(
  D.type({
    containerPort: Integer,
    image: D.type({
      name: NonEmptyString,
      tag: NonEmptyString,
    }),
    name: D.string,
    namespace: D.type({
      name: NonEmptyString,
    }),
    servicePort: Integer,
  }),
  D.intersect(
    D.partial({
      ingress: D.partial({
        secretName: D.string,
      }),
      labels: D.record(D.string),
      limits: D.type({ cpu: D.string, memory: D.string }),
      requests: D.type({ cpu: D.string, memory: D.string }),
    })
  )
);

export type AppComponentEnvironment = Omit<
  D.TypeOf<typeof AppComponentParams>,
  "namespace"
>;
export type Params = AppComponentEnvironment & GlobalEnvironment;
