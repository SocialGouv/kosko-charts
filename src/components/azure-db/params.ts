import {
  GlobalEnvironment,
  NamedComponentEnvironment,
} from "@socialgouv/kosko-charts/types";
import { NonEmptyString } from "@socialgouv/kosko-charts/utils/NonEmptyString";
import { pipe } from "fp-ts/lib/pipeable";
import * as D from "io-ts/lib/Decoder";

export const CreateDbComponentParams = pipe(
  D.type({
    database: NonEmptyString,
    password: NonEmptyString,
    user: NonEmptyString,
  }),
  D.intersect(
    D.partial({
      extensions: NonEmptyString,
    })
  )
);

export interface CreateDbJobParameters extends NamedComponentEnvironment {
  database: string;
  user: string;
  password: string;
  extensions?: string;
}

export type Params = CreateDbJobParameters & GlobalEnvironment;
