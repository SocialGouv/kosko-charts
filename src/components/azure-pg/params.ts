import type { PgParams } from "./types";

type PgDatabaseParams = Omit<PgParams, "host" | "name">;

export function getDevDatabaseParameters({
  suffix,
}: {
  suffix: string;
}): PgDatabaseParams {
  return {
    database: `autodevops_${suffix}`,
    password: `password_${suffix}`,
    user: `user_${suffix}`,
  };
}
