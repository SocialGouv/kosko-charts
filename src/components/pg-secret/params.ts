import {
  GlobalEnvironment,
  NamedComponentEnvironment,
} from "@socialgouv/kosko-charts/types";
import { NonEmptyString } from "@socialgouv/kosko-charts/utils/NonEmptyString";
import { pipe } from "fp-ts/lib/pipeable";
import * as D from "io-ts/lib/Decoder";

export const PostresSecretParameters = pipe(
  D.type({
    database: NonEmptyString,
    host: NonEmptyString,
    password: NonEmptyString,
    user: NonEmptyString,
  }),
  D.intersect(
    D.partial({
      sslmode: NonEmptyString,
    })
  )
);

export interface PostresSecretParameters extends NamedComponentEnvironment {
  database: string;
  user: string;
  password: string;
  host: string;
  sslmode?: string;
}

export type Params = PostresSecretParameters & GlobalEnvironment;
