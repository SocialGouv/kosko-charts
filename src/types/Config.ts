import { IIoK8sApimachineryPkgApisMetaV1ObjectMeta } from "kubernetes-models/_definitions/IoK8sApimachineryPkgApisMetaV1ObjectMeta";

interface Config {
  app: {
    metadata: IIoK8sApimachineryPkgApisMetaV1ObjectMeta;
    domain: string;
    subdomain: string;
  };
  namespace: { metadata: IIoK8sApimachineryPkgApisMetaV1ObjectMeta };
  registry: string;
}

export default Config;
