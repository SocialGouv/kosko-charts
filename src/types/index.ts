//

export interface GlobalEnvironment {
  annotations?: Record<string, string>;
  domain: string;
  ingress?: {
    annotations?: Record<string, string | undefined>;
  };
  labels?: Record<string, string>;
  namespace: {
    name: string;
  };
  subdomain: string;
}

export interface NamedComponentEnvironment {
  name: string;
}

export interface NamespaceComponentEnvironment {
  annotations?: Record<string, string>;
  labels?: Record<string, string>;
}
