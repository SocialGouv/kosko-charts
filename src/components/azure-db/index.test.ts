import { GlobalEnvironment } from "@socialgouv/kosko-charts/types";

import { create } from "./index";
import { CreateDbJobParameters, Params } from "./params";

test.each([
  ["because of missing variables", {} as Params],
  ["because of missing variables", { user: "zorro" } as Params],
  [
    "because of missing variables",
    { database: "", password: "", user: "" } as Params,
  ],
])("should throw %s", (_: string, params: Params) => {
  expect(() => create(params)).toThrowErrorMatchingSnapshot();
});

const validParams: CreateDbJobParameters = {
  database: "some-db",
  name: "test-job",
  password: "passw0rd",
  user: "tester",
};

const globalParams: GlobalEnvironment = {
  domain: "",
  namespace: { name: "sample-42-my-test" },
  subdomain: "",
};

test("should return a create-db job", () => {
  expect(
    create({
      ...globalParams,
      ...validParams,
    })
  ).toMatchSnapshot();
});

test("should use custom extensions", () => {
  expect(
    create({
      ...globalParams,
      ...validParams,
      extensions: "pgjwt",
    })
  ).toMatchSnapshot();
});
