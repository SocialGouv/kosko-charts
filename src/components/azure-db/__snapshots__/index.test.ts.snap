// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a create-db job 1`] = `
Object {
  "createDbJob": Object {
    "apiVersion": "batch/v1",
    "kind": "Job",
    "metadata": Object {
      "labels": Object {
        "app": "test-job",
      },
      "name": "test-job",
      "namespace": "sample-42-my-test",
    },
    "spec": Object {
      "backoffLimit": 0,
      "template": Object {
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "create-db-user",
              ],
              "env": Array [
                Object {
                  "name": "NEW_DB_NAME",
                  "value": "some-db",
                },
                Object {
                  "name": "NEW_USER",
                  "value": "tester",
                },
                Object {
                  "name": "NEW_PASSWORD",
                  "value": "passw0rd",
                },
                Object {
                  "name": "NEW_DB_EXTENSIONS",
                  "value": "hstore pgcrypto citext",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "azure-pg-admin-user",
                  },
                },
              ],
              "image": "registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:0.28.0",
              "imagePullPolicy": "IfNotPresent",
              "name": "create-db-user",
              "resources": Object {
                "limits": Object {
                  "cpu": "300m",
                  "memory": "256Mi",
                },
                "requests": Object {
                  "cpu": "100m",
                  "memory": "64Mi",
                },
              },
            },
          ],
          "restartPolicy": "Never",
        },
      },
    },
  },
}
`;

exports[`should throw because of missing variables 1`] = `
"BadArgument:required property \\"database\\"
└─ cannot decode undefined, should be string
required property \\"password\\"
└─ cannot decode undefined, should be string
required property \\"user\\"
└─ cannot decode undefined, should be string"
`;

exports[`should throw because of missing variables 2`] = `
"BadArgument:required property \\"database\\"
└─ cannot decode undefined, should be string
required property \\"password\\"
└─ cannot decode undefined, should be string"
`;

exports[`should throw because of missing variables 3`] = `
"BadArgument:required property \\"database\\"
└─ cannot decode \\"\\", should be NonEmptyString
required property \\"password\\"
└─ cannot decode \\"\\", should be NonEmptyString
required property \\"user\\"
└─ cannot decode \\"\\", should be NonEmptyString"
`;

exports[`should use custom extensions 1`] = `
Object {
  "createDbJob": Object {
    "apiVersion": "batch/v1",
    "kind": "Job",
    "metadata": Object {
      "labels": Object {
        "app": "test-job",
      },
      "name": "test-job",
      "namespace": "sample-42-my-test",
    },
    "spec": Object {
      "backoffLimit": 0,
      "template": Object {
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "create-db-user",
              ],
              "env": Array [
                Object {
                  "name": "NEW_DB_NAME",
                  "value": "some-db",
                },
                Object {
                  "name": "NEW_USER",
                  "value": "tester",
                },
                Object {
                  "name": "NEW_PASSWORD",
                  "value": "passw0rd",
                },
                Object {
                  "name": "NEW_DB_EXTENSIONS",
                  "value": "pgjwt",
                },
              ],
              "envFrom": Array [
                Object {
                  "secretRef": Object {
                    "name": "azure-pg-admin-user",
                  },
                },
              ],
              "image": "registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:0.28.0",
              "imagePullPolicy": "IfNotPresent",
              "name": "create-db-user",
              "resources": Object {
                "limits": Object {
                  "cpu": "300m",
                  "memory": "256Mi",
                },
                "requests": Object {
                  "cpu": "100m",
                  "memory": "64Mi",
                },
              },
            },
          ],
          "restartPolicy": "Never",
        },
      },
    },
  },
}
`;
